test_plans/atomic_ref.asciidoc

```diff
@@ -0,0 +1,471 @@
+:sectnums:
+:xrefstyle: short
+
+= Test plan for atomic_ref
+
+This is a test plan for the APIs described in SYCL 2020 section 4.15.3. Atomic references
+
+== Testing scope
+
+The test plan covers non-atomic behavior, the latter will be covered in a separate test plan
+
+=== Device coverage
+
+All of the tests described below are performed
+
+* on the default device that is selected on the CTS command line
+* on the host side
+
+== Type coverage
+
+All of the tests described below are performed using each of the
+following types as the `ReferencedType`.
+
+In Regular mode:
+
+* `int`
+* `float`
+
+In full conformance mode:
+
+* `int`
+* `unsigned int`
+* `long`
+* `unsigned long`
+* `float`
+
+In addition, if the device has `aspect::atomic64`, the following types are tested:
+
+* `long long`
+* `unsigned long long`
+* `double`
+
+Additionaly run same tests for `ReferencedType*`.
+
+Note: using T as `ReferencedType` or `ReferencedType*`.
+
+== Tests
+
+All non static tests run on following cases:
+
+* host code with referenced value is simple variable - should be skipped for `address_space::local_space`
```

Comment:
We have not yet decided which address spaces are legal when `atomic_ref` is used on the host.  In fact, it's possible that we will decide that `atomic_ref` isn't allowed in host code at all.  This is being discussed in the Khronos issue (https://gitlab.khronos.org/sycl/Specification/-/issues/637).

I think what you have here is fine for now, but can you add a note in this test plan with a pointer to that Khronos issue?  If this is still not resolved when you write the test, please add a comment in the test with a pointer to that issue, so we know to update the test when we resolve it.

Reply:
Thank you for information! That seems I don't have access to this issue, maybe there is some public tracker that can be added to the test plan for future update?

Reply:
The Khronos working group is still undecided about using `sycl::atomic_ref` in host code.  We plan to discuss it in the meeting on January 19, so I might have more information at that time.

If you want to make progress on this test, I suggest splitting it into two pieces.  The first part can test `sycl::atomic_ref` in device code.  This part is not controversial, so we can approve that plan whenever it is ready.  Once the working group decides about host code, we can either write a second test plan just for that case or we can update the test plan to include host testing.

Reply:
Got it, removed information concerning of testing `sycl::atomic_ref` on host side from this test plan. When the decision about using `sycl::atomic_ref` in host code have been done we update this test plan or add a new one for test case of using `sycl::atomic_ref` in host code.